### PR TITLE
new package: nfd

### PR DESCRIPTION
**Pull request by sjmackenzie**

### CODE REVIEW

**Comment by vcunat** [comment]
This needs a minor change to build on a chrooted machine (tried atop recent master):
```patch
--- a/pkgs/development/libraries/nfd/default.nix
+++ b/pkgs/development/libraries/nfd/default.nix
@@ -11,6 +11,7 @@ stdenv.mkDerivation {
   };
   buildInputs = [ doxygen boost pkgconfig openssl python pythonPackages.sphinx ndn-cxx graphviz libpcap];
   preConfigure = ''
+    patchShebangs ./waf
     ./waf configure \
       --boost-includes=${boost.dev}/include \
       --boost-libs=${boost.lib}/lib \
```

But the runtime config is still a problem, right? So this shouldn't be merged yet?

**Comment by iElectric** [comment]
ping :)

**Comment by sjmackenzie** [comment]
ah pong! Yes let me take a look at this tomorrow! Thanks!

**Comment by sjmackenzie** [comment]
travis said: 
```
[ 41/104] Compiling daemon/mgmt/internal-face.cpp

g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
```
Please could someone try this out on their localhost and merge.
Please note line 19 of the patch:
```
 --sysconfdir=$out/etc \
``` 
I presume using $out/etc suffices?

**Comment by vcunat** [comment]
I don't understand why that line is there. I presume typically one rather wants real ``/etc/*`` to be read (or have it configurable by an env var or a commandline parameter). BTW, the etc example is installed to ``$out/etc`` regardless of that line, although I don't know where the config is read from at runtime.

**Comment by sjmackenzie** [comment]
Okay the sysconfigdir is `$out/etc/ndn`.
I need to `cp $out/etc/ndn/ndn.conf.sample /etc/ndn/ndn.conf`
Then set 
```
./waf configure \
+ --boost-includes=${boost.dev}/include \
+ --boost-libs=${boost.lib}/lib \
+ --with-libpcap=${libpcap} \
+ --sysconfdir=/etc/ndn \
+ --prefix=$out
```
I do not have permission to create the `/etc/ndn` folder. 

Please advise.

**Comment by vcunat** [comment]
On nixpkgs level we do this by installing examples to ``$out/etc/``, but reading config from ``/etc/``.

Then on nixos one can then e.g. define ``environment.etc."ndn/ndn.conf"`` to get something on that path (either directly or a more comfortable nixos service might exist that creates the config and activates the service, like we have ``services.sshd.enable = true;``).

On non-nixos one needs to manage ``/etc/`` contents by hand.

**Comment by sjmackenzie** [comment]
Okay so something similar to this: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/cjdns.nix#L94 ? Is there a minimal service example/tutorial I could follow?

**Comment by vcunat** [comment]
I don't know. I don't see a tutorial for creating services in the official nixos manual. Maybe you can find something on the wiki or on one of many blog posts.

File: pkgs/development/libraries/nfd/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      --boost-libs=${boost.lib}/lib \
+      --with-libpcap=${libpcap} \
+      --sysconfdir=/etc/ndn \
+      --prefix=$out
+  '';
+  buildPhase = ''
+    ./waf
+  '';
+  installPhase = ''
+    mkdir /etc/ndn
```
Comment (bjornfor) [inline pkgs/development/libraries/nfd/default.nix]: nix packages cannot install things to /etc. Putting things in /etc is something that must be done in nixos modules.

Comment (sjmackenzie) [inline pkgs/development/libraries/nfd/default.nix]: okay great! Thanks for that!

File: pkgs/development/libraries/nfd/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{ stdenv, fetchgit, doxygen, boost, pkgconfig, python, pythonPackages, ndn-cxx, graphviz, libpcap, openssl }:
+let
+  version = "3651fd";
```
Comment (jagajaga) [inline pkgs/development/libraries/nfd/default.nix]: Change version from rev to date of that rev.

**Comment by jagajaga** [comment]
Ping? :)

**Comment by pSub** [comment]
Ping? :)

**Comment by domenkozar** [comment]
@sjmackenzie any plans to finish this package addition?

**Comment by sjmackenzie** [comment]
Ah no, we can kill it.

**Pull request closed without merging**